lib/vanagon/driver.rb

```diff
@@ -33,6 +37,18 @@ def initialize(platform, project, options = { :configdir => nil, :target => nil,
       raise Vanagon::Error.wrap(e, "Could not load the desired engine '#{engine}'")
     end
 
+    def filter_out_components(only_build)
+      filtered_components = []
+      only_build.each do |comp|
+        filtered_components += @project.filter_component(comp)
+      end
```

Comment:
No idea where the last comment I left went, but the gist of it is that we can boil this down some using `Enumerable#map` here:

```ruby
filtered_components = only_build.map { |comp| @project.filter_component(comp) }.uniq
```